File: leetcode/src/scala/ReverseWordsInAString.scala
object ReverseWordsInAString {
    def reverseWords(s: String): String = {
        var result = ""
        val arr = s.split(" +")
        Array.range(arr.length - 1, -1, -1).foreach { i => if (arr(i).nonEmpty) result += " " + arr(i)}
        result = result.substring(1)
        result
    }

    def main(args: Array[String]): Unit = {
        val s = "the sky is blue"
        println(reverseWords(s))
    }
}

File: leetcode/src/python/reverse_words_in_a_string.py
class ReverseWordsInAString:
    def reverseWords(self, s: str) -> str:
        result = ''
        arr = s.split()
        for i in range(len(arr) - 1, -1, -1):
            if (arr[i] != ''):
                result += ' ' + arr[i]
        result = result[1:]
        return result